modmesh/app/euler1d.py

```diff
@@ -142,215 +301,72 @@ def __init__(self, shocktube):
-        therefore the checked state is not used in this function.
+        Initializing solver configure by user's input, also reset
+        the computational results.
 
-        :param checked: button is checked or not
-        :return: nothing
+        :return None
         """
-        self.use_grid_layout = not self.use_grid_layout
-        self._update_layout()
-
-    def delete_layout(self, layout):
-        """Delete all widgets from a layout."""
-        for i in reversed(range(layout.count())):
-            widget = layout.itemAt(i).widget()
-            # Check if the widget is a layout
-            if isinstance(widget, QLayout):
-                self.delete_layout(widget)
-            else:
-                widget.setParent(None)
-                widget.deleteLater()
-
-        layout.update()
-        layout.setParent(None)
-        layout.deleteLater()
-
-    def eventFilter(self, obj, event):
-        if self.use_grid_layout:
-            if event.type() == QEvent.MouseButtonPress:
-                self.mousePressEvent(event)
-            elif event.type() == QEvent.MouseMove:
-                self.mouseMoveEvent(event)
-
-        return super().eventFilter(obj, event)
-
-    def mousePressEvent(self, event):
-        if event.button() == Qt.LeftButton and self.use_grid_layout:
-            coord = event.windowPos().toPoint()
-            self.targetIndex = self.getWindowIndex(coord)
-        else:
-            self.targetIndex = None
-
-    def mouseMoveEvent(self, event):
-        if event.buttons() == Qt.LeftButton and self.targetIndex is not None:
-            windowItem = self.figure_layout.itemAt(self.targetIndex)
-
-            drag = QDrag(windowItem)
-
-            pix = windowItem.itemAt(0).widget().grab()
-
-            mimeData = QMimeData()
-            mimeData.setImageData(pix)
-
-            drag.setMimeData(mimeData)
-            drag.setPixmap(pix)
-            drag.setHotSpot(event.pos())
-            drag.exec()
-
-    def dragEnterEvent(self, event):
-        if event.mimeData().hasImage():
-            event.accept()
-        else:
-            event.ignore()
-
-    def dropEvent(self, event):
-        if not event.source().geometry().contains(event.pos()):
-            targetWindowIndex = self.getWindowIndex(event.pos())
-            if targetWindowIndex is None:
-                return
-
-            i, j = (max(self.targetIndex, targetWindowIndex),
-                    min(self.targetIndex, targetWindowIndex))
-
-            p1, p2 = (self.figure_layout.getItemPosition(i),
-                      self.figure_layout.getItemPosition(j))
-
-            self.figure_layout.addItem(self.figure_layout.takeAt(i), *p2)
-            self.figure_layout.addItem(self.figure_layout.takeAt(j), *p1)
-
-    def getWindowIndex(self, pos):
-        for i in range(self.figure_layout.count()):
-            if self.figure_layout.itemAt(i).geometry().contains(pos):
-                return i
```

Comment:
Removing drag and drop reordering feature, it would be reimplemented within the configuration window of the plotting area, but the implementation way may not using drag and drop.